language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/completions/builder/FunctionCompletionItemBuilder.java

```diff
@@ -373,6 +378,40 @@ private static String getQualifiedFunctionName(String functionName, BallerinaCom
         return modulePrefix + SyntaxKind.COLON_TOKEN.stringValue() + functionName;
     }
 
+    /**
+     * Creates and returns the main function completion item.
+     *
+     * @param context   Ballerina completion context
+     * @return {@link CompletionItem} generated main function completion item.
+     */
+    public static CompletionItem buildMainFunction(BallerinaCompletionContext context) {
+        NonTerminalNode node = context.getNodeAtCursor();
+        Optional<Token> lastQualifier = Optional.empty();
+        while (node != null) {
+            lastQualifier = CommonUtil.getLastQualifier(context, node);
+            if (lastQualifier.isPresent() || node.kind() == SyntaxKind.MODULE_PART) {
+                break;
+            }
+            node = node.parent();
+        }
+
+        CompletionItem completionItem = new CompletionItem();
+        String insertText;
+        if (lastQualifier.isPresent() && lastQualifier.get().text().contains(ItemResolverConstants.PUBLIC_KEYWORD)) {
```

Comment:
We can check `lastQualifier.get().kind() == SyntaxKind.PUBLIC_KEYWORD` instead of doing a string contains check

Reply:
done